### ISSUE TITLE
Git Sync behaviour is broken

### ISSUE THREAD

**Issue by shubhendu-enable** [getinsomnia/insomnia#8043]
### Expected Behavior

I should be able to pull, push , and resolve conflicts if there are any.

### Actual Behavior

It's showing uncommitted changes when i see in document

<img width="1061" alt="image" src="https://github.com/user-attachments/assets/a01f2f18-1395-44ac-a8a4-7b971b9e0880">
when i try to see changes , it shows no changes

<img width="790" alt="image" src="https://github.com/user-attachments/assets/5badd3c4-09f7-487d-b90e-a55c3f1724ad">

when i tried to pull, it shows no conflict supported

<img width="782" alt="image" src="https://github.com/user-attachments/assets/fc5b6825-42fd-46e9-aa77-8da75952ef16">

So basically, I am blocked as I can not do any actions , Please suggest how to fix this.

### Reproduction Steps

1. Sync Document with git repository
2. Add API Spec
3. Add API collections
4. Add Tests

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Insomnia Version

10.1.0-beta.2

### What operating system are you using?

macOS

### Operating System Version

15.0.1

### Installation method

download from insomnia website

### Last Known Working Insomnia version

NA

**Comment by shubhendu-enable** [comment]
Also getting error when tried to fetch
An internal error caused this command to fail. Please file a bug report at https://github.com/isomorphic-git/isomorphic-git/issues with this error message: Index file is empty (.git/index)

**Comment by shubhendu-enable** [comment]
When i dropped the document and tried to resync, it loaded very slowly, and after making some changes, when I tried to revert the changes, the Rollback action was not working

<img width="812" alt="image" src="https://github.com/user-attachments/assets/dd1f4c04-4de0-44f4-a9ee-420ddec72b91">

**Comment by subnetmarco** [comment]
We are about to push an large amount of improvements on Git Sync next week in v10.1.

cc @gatzjames

**Comment by gatzjames** [comment]
Hey @shubhendu-enable we have made some significant updates to the Git UX in the app in the latest beta if you want to try it out and let us know how it works for you.
It will be available soon in the next patch release

**Comment by shubhendu-enable** [comment]
@gatzjames 
I upgraded to Version: Insomnia 10.1.0
Some observations on git sync behaviour:
1. First, the UI interface for git commit is much better now, Thanks for updating this commit experience.
2. Performance perspective git sync is still slow and has some unexpected behavior, 
   - Once I click on commit changes - it takes between 1-2 min to load changes and shows Stage Changes and Changes 
   - If I click on any of the changes to see the difference, it again takes around 1 min to see the difference 
   - Next, i want to stage some changes, this process also takes around 1 min to reflect on the staged changes section and in between this process other actions on the screen became unresponsive
   - I tried to rollback some changes from the Changes Section, but instead of rolling back that change, the application routed back to Collections.
